scheduler/scheduler_shuffle.go

```diff
@@ -45,6 +45,7 @@ func (s *syncClient) syncShuffle(ctx context.Context, resolvedResources chan<- *
 	// so users have a little bit of control over the randomization.
 	seed := hashTableNames(tableNames)
 	shuffle(tableClients, seed)
+	tableClients = shardTableClients(tableClients, s.shard)
```

Comment:
I do see that `shuffle` is deterministic (at the moment), but I still think it's a bad idea to shard after shuffling. I'd move it before the shuffle.

Reply:
OK let me try and switch the order and re-run the tests. We shuffle (this is the default in AWS) to avoid rate limits. Don't think sharding before shuffling will make a difference in that aspect but I'll re-test

Reply:
> Don't think sharding before shuffling will make a difference in that aspect but I'll re-test

I think it will be fine since we round-robin before we shuffle anyways

Reply:
Ok did a bit of testing and it looks good so we can shard before shuffle

Reply:
Is there any case where collecting the tables could be non-deterministic? Normally the tables are hardcoded in a plugin, so it should not be the case. If there is a plugin where the tables are dynamic, and they could change between syncs (e.g. if they are discovered by an API which is non-deterministic), sharding would not work.

In either case, I think the deterministic requirement is worth a one-liner comment.

Reply:
> Is there any case where collecting the tables could be non-deterministic?

This is a good point, and definitely a limitation of this approach, see below ⬇️ 

- It can happen due to a bug in the plugin https://github.com/cloudquery/cloudquery-private/pull/4299.
- Plugins with dynamic tables don't use the scheduler, they do their own thing so they would need to implement sharding on the plugin's side (if needed. e.g. For Postgres source probably better to use a stronger machine instead of sharding)
- I can think of other cases, e.g. some creating an AWS account after shard 1/2 discovery and before shard 2/2 discovery. If we discover all accounts, that will mess up the sharding. A solution would be to hard code the accounts in the spec to avoid it.